src/discussions/discussions-home/DiscussionSidebar.jsx

```diff
@@ -1,17 +1,27 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
 
 import classNames from 'classnames';
+import { useSelector } from 'react-redux';
 import {
   Redirect, Route, Switch, useLocation,
 } from 'react-router';
 
-import { Routes } from '../../data/constants';
+import { AppContext } from '@edx/frontend-platform/react';
+
+import { RequestStatus, Routes } from '../../data/constants';
 import { PostsView } from '../posts';
+import {
+  selectAllThreads, threadsLoadingStatus,
+} from '../posts/data/selectors';
 import { TopicsView } from '../topics';
 
 export default function DiscussionSidebar({ displaySidebar }) {
   const location = useLocation();
+  const AllThreads = useSelector(selectAllThreads);
+  const { authenticatedUser } = useContext(AppContext);
+  const userThreads = AllThreads.filter(thread => thread.author === authenticatedUser.username);
```

Comment:
I don't think this is the right approach. The user could have threads that are not on the first page, and this will be empty. I'm not sure why you changed the logic here.

Reply:
`selectAllThreads` is returning all the threads but because the default filter is `author: authenticatedUser.username` , it returns the `userThreads`. With the test it was not working as expected.

Reply:
That is because the test is incorrect, not the code. Please fix the test. 
In your test, the API is returning the same results whether you filter by user or not. So, the correct thing to do is to override the results returned based on the API params.